### PR TITLE
CRM: Adding remaining Automation Contact Actions

### PR DESCRIPTION
**Pull request by coder-karen**
## Proposed changes:

* This PR adds the remaining Automation Contact actions - 'update contact', 'new contact', 'add / remove contact tags', 'add contact log' actions). It also updates the name of the 'update contact' action defined in the initial PR - https://github.com/Automattic/jetpack/pull/31109 - to 'update contact status' to correctly reflect what it is doing.
* This PR also includes tests for those actions.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

https://github.com/Automattic/zero-bs-crm/issues/3125

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* Run tests by navigating to the CRM directory in your local installation after running `jetpack build plugins/crm` with this PR applied, and then run `composer phpunit -- --testdox --testsuite automation` (note this will currently run all Automation tests)
* To test just one: `composer phpunit -- --filter test_add_remove_contact_tag_action_with_workflow tests/php/automation/contacts/class-contact-action-test.php --testdox --testsuite automation` (replacing the test name as needed)

### CODE REVIEW

File: projects/plugins/crm/src/automation/commons/actions/contacts/class-add-contact-log.php [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	/**
+	 * Add the log to the contact via the DAL.
+	 *
+	 * @param array $contact_data The contact data on which the log is to be added.
+	 */
+	public function execute( array $contact_data = array() ) {
+		global $zbs;
+
+		$contact_data = $this->attributes;
+		$zbs->DAL->contacts->zeroBS_addUpdateObjLog( $contact_data ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
```
Comment (cleacos) [inline projects/plugins/crm/src/automation/commons/actions/contacts/class-add-contact-log.php]:
```suggestion
		$zbs->DAL->contacts->zeroBS_addUpdateObjLog( $this->attributes ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
```

File: projects/plugins/crm/src/automation/commons/actions/contacts/class-add-remove-contact-tag.php [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	/**
+	 * Add / remove the tag to / from the contact via the DAL.
+	 *
+	 * @param array $contact_data The contact data on which the tag is to be added / removed.
+	 */
+	public function execute( array $contact_data = array() ) {
+		global $zbs;
+
+		$contact_data = $this->attributes;
+		$zbs->DAL->contacts->addUpdateContactTags( $contact_data ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
```
Comment (cleacos) [inline projects/plugins/crm/src/automation/commons/actions/contacts/class-add-remove-contact-tag.php]:
```suggestion
		$zbs->DAL->contacts->addUpdateContactTags( $this->attributes ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
```

File: projects/plugins/crm/src/automation/commons/actions/contacts/class-new-contact.php [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+	/**
+	 * Add the new contact to the DAL.
+	 *
+	 * @param array $contact_data The contact data to be added.
+	 */
+	public function execute( array $contact_data = array() ) {
+		global $zbs;
+
+		$contact_data = $this->attributes;
+		$zbs->DAL->contacts->addUpdateContact( $contact_data ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
```
Comment (cleacos) [inline projects/plugins/crm/src/automation/commons/actions/contacts/class-new-contact.php]:
```suggestion
		$zbs->DAL->contacts->addUpdateContact( $this->attributes ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
```

File: projects/plugins/crm/src/automation/commons/actions/contacts/class-add-contact-log.php (hunk 1 above)
Comment (coder-karen) [reply projects/plugins/crm/src/automation/commons/actions/contacts/class-add-contact-log.php]: Thanks!

**Review by cleacos — APPROVED** [review]
Test passed. The code looks good.

LGTM!

**Pull request merged**